src/Bootstrappers/PrefixCacheTenancyBootstrapper.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers;
+
+use Closure;
+use Illuminate\Cache\CacheManager;
+use Illuminate\Cache\Repository;
+use Illuminate\Contracts\Config\Repository as ConfigRepository;
+use Illuminate\Support\Facades\Cache;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class PrefixCacheTenancyBootstrapper implements TenancyBootstrapper
+{
+    protected array $originalPrefixes = []; // E.g. 'redis' => 'redis_prefix_'
+    public static array $tenantCacheStores = []; // E.g. 'redis'
+    public static array $prefixGenerators = [
+        // driverName => Closure(Tenant $tenant)
+    ];
+
+    public function __construct(
+        protected ConfigRepository $config,
+        protected CacheManager $cacheManager,
+    ) {
+    }
+
+    public function bootstrap(Tenant $tenant): void
+    {
+        $originalPrefix = $this->config->get('cache.prefix');
+
+        foreach (static::$tenantCacheStores as $store) {
+            $this->originalPrefixes[$store] = $originalPrefix;
+
+            $this->setCachePrefix($store, $this->getStorePrefix($store, $tenant));
+        }
+    }
+
+    public function revert(): void
+    {
+        foreach ($this->originalPrefixes as $driver => $prefix) {
+            $this->setCachePrefix($driver, $prefix);
+        }
+
+        $this->originalPrefixes = [];
+    }
+
+    protected function setCachePrefix(string $driver, string|null $prefix): void
+    {
+        $this->config->set('cache.prefix', $prefix);
+
+        // Refresh driver's store to make the driver use the current prefix
+        $this->refreshStore($driver);
+
+        // It is needed when a call to the facade has been made before bootstrapping tenancy
+        // The facade has its own cache, separate from the container
+        Cache::clearResolvedInstances();
+    }
+
+    public function getStorePrefix(string $store, Tenant $tenant): string
+    {
+        if (isset(static::$prefixGenerators[$store])) {
+            return static::$prefixGenerators[$store]($tenant);
+        }
+
+        return $this->originalPrefixes[$store] . config('tenancy.cache.prefix_base') . $tenant->getTenantKey();
```

Comment:
We can use `$this->config` instead of `config()`

Reply:
Right, changed that